internal/kafka/test/integration/kafka_expiration_test.go

```diff
@@ -0,0 +1,513 @@
+							),
+						},
+						SupportedBillingModels: []config.KafkaBillingModel{
+							{
+								ID:               types.STANDARD.String(),
+								AMSResource:      ocm.RHOSAKResourceName,
+								AMSProduct:       string(ocm.RHOSAKTrialProduct),
+								AMSBillingModels: []string{string(amsv1.BillingModelStandard)},
+							},
+						},
+					},
+				},
+			},
+		}
+	})
+	defer tearDown()
+
+	mockKasFleetshardSyncBuilder := kasfleetshardsync.NewMockKasFleetshardSyncBuilder(h, t)
+	mockKasfFleetshardSync := mockKasFleetshardSyncBuilder.Build()
+	mockKasfFleetshardSync.Start()
+	defer mockKasfFleetshardSync.Stop()
+
+	// set up dummy data plane cluster that accepts both standard and developer clusters
+	testCluster := mockclusters.BuildCluster(func(cluster *api.Cluster) {
+		cluster.Meta = api.Meta{
+			ID: api.NewID(),
+		}
+		cluster.ProviderType = api.ClusterProviderStandalone // ensures no errors will occur due to it not being available on ocm
+		cluster.SupportedInstanceType = api.AllInstanceTypeSupport.String()
+		cluster.ClientID = "some-client-id"
+		cluster.ClientSecret = "some-client-secret"
+		cluster.ClusterID = "test-cluster"
+		cluster.CloudProvider = mocks.MockCloudProviderID
+		cluster.Region = mocks.MockCloudRegionID
+		cluster.Status = api.ClusterReady
+		cluster.ProviderSpec = api.JSON{}
+		cluster.ClusterSpec = api.JSON{}
+		cluster.AvailableStrimziVersions = api.JSON(mockclusters.AvailableStrimziVersions)
+	})
+	db := h.DBFactory().New()
+	err := db.Create(testCluster).Error
+	g.Expect(err).NotTo(gomega.HaveOccurred(), "failed to create dummy data plane cluster")
+
+	// ensure kafka request with long lifespan does not get deleted
+	account := h.NewRandAccount()
+	ctx := h.NewAuthenticatedContext(account, nil)
+	result, resp, err := client.DefaultApi.CreateKafka(ctx, true, public.KafkaRequestPayload{
+		Name: "kafka-with-long-lifespan",
+		Plan: fmt.Sprintf("%s.x1", types.STANDARD), // has a long lifespan of 2d
+	})
+	if resp != nil {
+		_ = resp.Body.Close()
+	}
+	g.Expect(err).ToNot(gomega.HaveOccurred(), "failed to create kafka with long lifespan")
+	expectedExpirationDate := result.CreatedAt.Add(time.Duration(longLifespanSeconds) * time.Second).Format(time.RFC1123Z)
+	g.Expect(result.ExpiresAt.Format(time.RFC1123Z)).To(gomega.Equal(expectedExpirationDate), "expected expires_at for kafka with long lifespan does not match actual")
+
+	// wait for kafka to be ready and ensure the expiration date does not get updated
+	kafka, err := common.WaitForKafkaToReachStatus(ctx, test.TestServices.DBFactory, client, result.Id, constants.KafkaRequestStatusReady)
+	g.Expect(err).ToNot(gomega.HaveOccurred(), "failed to wait for kafka with long lifespan to reach status %q", constants.KafkaRequestStatusReady)
+	g.Expect(kafka.ExpiresAt.Format(time.RFC1123Z)).To(gomega.Equal(expectedExpirationDate), "expected expires_at for kafka with long lifespan does not match actual")
```

Comment:
Why is this compared again?

Reply:
It's to ensure that the expiration date never gets updated. We expect the expiration date to be set on creation. KFM should not set it again during reconcile, this verifies that.

Reply:
:+1: